File: app/src/main/java/io/shelfy/architecture/data/datasource/remote/mappers/MovieMapper.java
package io.shelfy.architecture.data.datasource.remote.mappers;

import androidx.annotation.NonNull;

import io.shelfy.architecture.ApiConstants;
import io.shelfy.architecture.data.datasource.remote.response.MovieJson;
import io.shelfy.architecture.common.Mapper;
import io.shelfy.architecture.domain.entity.Movie;

public class MovieMapper implements Mapper<MovieJson, Movie> {

    @NonNull
    @Override
    public Movie map(@NonNull MovieJson movieJson) {
        return new Movie(
                movieJson.getId(),
                movieJson.getTitle(),
                movieJson.getOverview(),
                movieJson.getReleaseDate(),
                ApiConstants.POSTER_BASE_URL + movieJson.getPosterPath(),
                ApiConstants.BACKDROP_BASE_URL + movieJson.getBackdropPath()
        );
    }
}

File: app/src/main/java/io/shelfy/architecture/data/mappers/MovieMapper.kt
package io.shelfy.architecture.data.mappers

import io.shelfy.architecture.api.ApiConstants
import io.shelfy.architecture.api.response.MovieJson
import io.shelfy.architecture.domain.model.Movie
import io.shelfy.architecture.domain.mappers.Mapper

class MovieMapper : Mapper<MovieJson, Movie> {

    override fun map(from: MovieJson): Movie {

        return with(from) {
            Movie(
                id = id,
                title = title,
                description = overview,
                releaseDate = releaseDate,
                posterUrl = ApiConstants.POSTER_BASE_URL + posterPath,
                backdropUrl = ApiConstants.BACKDROP_BASE_URL + backdropPath
            )
        }
    }
}
